.github/actions/install-requirement/action.yml

```diff
@@ -0,0 +1,51 @@
+name: Install Project Requirements
+description: Install requirements from a PEP 517 project
+
+inputs:
+  requirements:
+    description: "Whitespace-delimited list of project requirements to install."
+    required: true
+  project-root:
+    description: "The filesystem path to the root of the project to install from; defaults to github.workspace."
+    default: ${{ github.workspace }}
+  extra:
+    description: "Name of the optional dependencies marker; e.g. dev."
+    required: false
+    default: ""
```

Comment:
Checking that I'm reading this right - if you *don't* specify extra as an argument, this manifests as `--extra ""` in the final arguments, right? Would it be worth gating the `--extra` argument with an `if inputs.extra`?

Reply:
The default value of `extra` within the `install_requirement.py` script is an empty string. So, passing `--empty` an empty string at the command-line is equivalent to not specifying `--extra` at all. With that, we _can_ control specifying `--extra` at the command-line but it's effectively duplicating effort the script already handles.

Reply:
That said...

```diff
         python -m install_requirement ${{ inputs.requirements }} \
-          --extra "${{ inputs.extra }}" \
+          ${{ inputs.extra && format('--extra {0}', inputs.extra) || '' }} \
           --project-root "$(printf -- "%q" "${{ steps.paths.outputs.project-root }}")"
```